repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java

```diff
@@ -1823,7 +1823,7 @@ private boolean shouldDeleteExistingRelations(AttributeMutationContext ctx, Atla
         boolean ret = false;
 
         AtlasEntityType entityType = typeRegistry.getEntityTypeByName(AtlasGraphUtilsV2.getTypeName(ctx.getReferringVertex()));
-        if (entityType.hasRelationshipAttribute(attribute.getName())) {
+        if (entityType !=null && entityType.hasRelationshipAttribute(attribute.getName())) {
```

Comment:
is it possible to link the same struct vertex to different entity vertices?
I guess a struct vertex will only be related to a single vertex 
Just confirming...

Reply:
> is it possible to link the same struct vertex to different entity vertices?

No, Every Entity vertex will have a its edge to struct vertex which is created while entity creation. its 1 to 1 mapping

> I guess a struct vertex will only be related to a single vertex

Yes